File: FunctionalBlazor/FunctionalBlazor.Program/Startup.fs
namespace FunctionalBlazor.Program

open Microsoft.Extensions.DependencyInjection
open ProgramTypes
open Program
open System.Runtime.CompilerServices

[<Extension>]
type Startup  () =
        
    [<Extension>]
    static member inline ConfigureProgram (services : IServiceCollection) =
        services.AddScoped<IProgram, Program>() |> ignore

File: FunctionalBlazor/FunctionalXamarin/Startup.cs
using Microsoft.Extensions.DependencyInjection;
using System;
using System.Collections.Generic;
using System.Text;

namespace FunctionalXamarin
{
    public static class Startup
    {
        public static void ConfigureViewModels(this IServiceCollection services)
        {
            services.AddScoped<CounterViewModel>();
        }
    }
}
